aptos-move/framework/aptos-framework/sources/TokenV1.move

```diff
@@ -0,0 +1,935 @@
+    const ECOLLECTIONS_NOT_PUBLISHED: u64 = 2;
+    const ECOLLECTION_NOT_PUBLISHED: u64 = 3;
+    const ECOLLECTION_ALREADY_EXISTS: u64 = 4;
+    const ECREATE_WOULD_EXCEED_MAXIMUM: u64 = 5;
+    const EINSUFFICIENT_BALANCE: u64 = 6;
+    const EINVALID_COLLECTION_NAME: u64 = 7;
+    const EINVALID_TOKEN_MERGE: u64 = 8;
+    const EMINT_WOULD_EXCEED_MAXIMUM: u64 = 9;
+    const ENO_BURN_CAPABILITY: u64 = 10;
+    const ENO_MINT_CAPABILITY: u64 = 11;
+    const ETOKEN_ALREADY_EXISTS: u64 = 12;
+    const ETOKEN_NOT_PUBLISHED: u64 = 13;
+    const ETOKEN_STORE_NOT_PUBLISHED: u64 = 14;
+
+    //
+    // Core data structures for holding tokens
+    //
+
+    struct Token has store {
+        id: TokenId,
+        value: u64,
+    }
+
+    struct TokenId has store, copy, drop {
+        tokendata_id: TokenDataId,
+        serial_number: u64
+    }
+
+    struct TokenDataId has copy, drop, store {
+        // The creator of this token
+        creator: address,
+        // The collection or set of related tokens within the creator's account
+        collection: String,
+        // Unique name within a collection within the creator's account
+        name: String,
+    }
+
+    /// The royalty of a token
+    struct Royalty has copy, drop, store {
+        royalty_points_nominator: u64,
+        royalty_points_denominator: u64,
+        // if the token is jointly owned by multiple creators,
+        // the group of creators should create a shared account.
+        // the creator_account will be the shared account address.
+        resource_account: address,
+    }
+
+    // This config specifies whether fields in the Collection are mutable
+    struct CollectionMutabilityConfig has copy, store, drop {
+        description: bool,
+        // URL for additional information /media
+        uri: bool,
+        // Optional maximum number of tokens allowed within this collections
+        maximum: bool,
+    }
+
+    // This config specifies whether fields in the TokenData are mutable
+    struct TokenMutabilityConfig has copy, store, drop {
+        maximum: bool,
+        uri: bool,
+        royalty: bool,
+        description: bool,
+        default_property_list: bool,
+    }
+
+    struct PropertyValue has store, copy, drop {
+        type: String,
+        value: vector<vector<u8>>,
+    }
+
+    /// The data associated with the Tokens
+    struct TokenData has store {
+        id: TokenDataId,
+        maximum: u64,
+        // Total number of this type of Token
+        supply: u64,
+        // The max serial number we should have for the token
+        largest_serial_number: u64,
+        // URL for additional information / media
+        uri: String,
+        // the royalty of the token
+        royalty: Royalty,
+        // The name of this Token
+        name: String,
+        // Describes this Token
+        description: String,
+        // the properties can be used to hold any custom attributes
+        default_property_list: PropertyList,
```

Comment:
I thought we were going to use SimpleMap?
Also `properties`
and `SimpleMap<String, vec<u8>>` -- lots of feedback from folks that they want unicode.

Reply:
The propertylist uses simpleMap<String, PropertyValue>. PropertyValue {type: String, value: vector<u8>}. The value is vector<u8>.